src/pages/settings/Profile/CustomStatus/StatusPage.js

```diff
@@ -50,7 +50,16 @@ function StatusPage({draftStatus, currentUserPersonalDetails}) {
         User.clearDraftCustomStatus();
     };
 
-    const navigateBackToSettingsPage = useCallback(() => Navigation.goBack(ROUTES.SETTINGS_PROFILE, false, true), []);
+    const topMostReportID = Navigation.getTopmostReportId();
+
+    const navigateBackToSettingsPage = useCallback(() => {
+        if (topMostReportID) {
+            Navigation.goBack(ROUTES.REPORT_WITH_ID.getRoute(topMostReportID));
+        } else {
+            Navigation.goBack(ROUTES.SETTINGS_PROFILE, false, true);
+        }
+    }, [topMostReportID]);
```

Comment:
Should we take topMostReportID inside the navigateBackToSettingsPage function?

Reply:
Maybe I haven't used `useCallback` much recently, but if we put `Navigation.getTopmostReportId();` inside `navigateBackToSettingsPage`, won't `getTopmostReportId` always get the most up to date `topMostReportID` 
since `getTopmostReportId` will be called right when the value is needed?